app/models/bookings/school.rb

```diff
@@ -2,9 +2,55 @@ class Bookings::School < ApplicationRecord
   include FullTextSearch
   include GeographicSearch
 
-  validates :name, presence: true, length: { maximum: 128 }
+  validates :name,
+    presence: true,
+    length: { maximum: 128 }
 
-  def self.table_name_prefix
-    'bookings_'
+  validates :fee,
+    presence: true,
+    numericality: { greater_than_or_equal_to: 0, only_integer: true }
+
+  has_many :bookings_schools_subjects,
+    class_name: "Bookings::SchoolsSubject",
+    inverse_of: :bookings_school,
+    foreign_key: :bookings_school_id
+
+  has_many :subjects,
+    class_name: "Bookings::Subject",
+    through: :bookings_schools_subjects,
+    source: :bookings_subject
+
+  has_many :bookings_schools_phases,
+    class_name: "Bookings::SchoolsPhase",
+    inverse_of: :bookings_school,
+    foreign_key: :bookings_school_id
+
+  has_many :phases,
+    class_name: "Bookings::Phase",
+    through: :bookings_schools_phases,
+    source: :bookings_phase
+
+  scope :that_provide, ->(subject_ids) do
+    if subject_ids.present?
+      joins(:subjects).merge(Bookings::Subject.where(id: subject_ids)).distinct
+    else
+      all
+    end
+  end
+
+  scope :at_phases, ->(phase_ids) do
+    if phase_ids.present?
+      joins(:phases).merge(Bookings::Phase.where(id: phase_ids)).distinct
```

Comment:
Not really questioning this but wanting to understand the logic behind using merge verses and chained where, ie

```ruby
joins(:phases).where(bookings_phases: { id: phase_ids })
```

Does this use the appropriate table name irrespective of the aliasing from the join?

Reply:
I didn't realise I could do that directly across `has_many through:` relationships directly without nesting, thanks!